### PR TITLE
mgr/cephadm: upgrade requires root mode for now

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2785,7 +2785,18 @@ def upgrade_status(self):
...
+                self.mode))
         if version:
+            try:
+                (major, minor, patch) = version.split('.')
+                assert int(minor) >= 0
+                assert int(patch) >= 0
+            except:
+                raise OrchestratorError('version must be in the form X.Y.Z (e.g., 15.2.3)')
+            if int(major) < 15 or (int(major) == 15 and int(minor) < 2):
+                raise OrchestratorError('cephadm only supports octopus (15.2.0) or later')
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py]: how will this work for the monitoring components (node-exporter, prom, grafana)?

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -2785,7 +2785,18 @@ def upgrade_status(self):
...
 
     def upgrade_start(self, image, version):
+        if self.mode != 'root':
+            raise OrchestratorError('upgrade is not supported in %s mode' % (
+                self.mode))
         if version:
+            try:
+                (major, minor, patch) = version.split('.')
+                assert int(minor) >= 0
+                assert int(patch) >= 0
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py]: make this less restrictive? currently only major/minor is reported for nfs-ganesha https://github.com/ceph/ceph/pull/33064/commits/d4fabe294f4031d89d2bca067dc6056a4c5f89f7

File: src/pybind/mgr/cephadm/module.py (hunk 1 above)
Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: it wont.  this only supports upgrading ceph (currently)

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: these are only ceph versions

**Pull request merged**